tools/pyinstaller_hooks/hook-raiden.py

```diff
@@ -17,8 +17,13 @@ def copy_metadata(package_name):
 
 # Add metadata of all required packages to allow pkg_resources.require() to work
 required_packages = [("raiden", [])]
+processed_packages = set()  # break out of circular dependencies
 while required_packages:
     req_name, req_extras = required_packages.pop()
+    if req_name in processed_packages:
+        continue
+    processed_packages.add(req_name)
```

Comment:
I'd suggest to move the check down into the for loop and including both the requirement and its extras in the set. Otherwise in cases where a dependency is included multiple times but with different extras we would miss those.

Reply:
Is that even allowed? I thought having a dependency multiple times with different extras will cause a dependency resolution error, just like when you require the same dependency in multiple different versions.

Reply:
I was thinking of a sub dependency in two unrelated packages.

Reply:
I'm also not sure if that is allowed. Anyway, I changed the code as requested.